tests/test_param_types.py

```diff
@@ -0,0 +1,106 @@
+import jax
+import pytest
+
+from algorithmic_efficiency.workloads.cifar.cifar_jax.workload import \
+    CifarWorkload as JaxCifarWorkload
+from algorithmic_efficiency.workloads.cifar.cifar_pytorch.workload import \
+    CifarWorkload as PyTorchCifarWorkload
+from algorithmic_efficiency.workloads.criteo1tb.criteo1tb_jax.workload import \
+    Criteo1TbDlrmSmallWorkload as JaxCriteoWorkload
+from algorithmic_efficiency.workloads.criteo1tb.criteo1tb_pytorch.workload import \
+    Criteo1TbDlrmSmallWorkload as PyTorchCriteoWorkload
+from algorithmic_efficiency.workloads.imagenet_resnet.imagenet_jax.workload import \
+    ImagenetResNetWorkload as JaxImagenetResNetWorkload
+from algorithmic_efficiency.workloads.imagenet_resnet.imagenet_pytorch.workload import \
+    ImagenetResNetWorkload as PyTorchImagenetResNetWorkload
+from algorithmic_efficiency.workloads.imagenet_vit.imagenet_jax.workload import \
+    ImagenetVitWorkload as JaxImagenetViTWorkload
+from algorithmic_efficiency.workloads.imagenet_vit.imagenet_pytorch.workload import \
+    ImagenetVitWorkload as PyTorchImagenetViTWorkload
+from algorithmic_efficiency.workloads.mnist.mnist_jax.workload import \
+    MnistWorkload as JaxMnistWorkload
+from algorithmic_efficiency.workloads.mnist.mnist_pytorch.workload import \
+    MnistWorkload as PyTorchMnistWorkload
+from algorithmic_efficiency.workloads.ogbg.ogbg_jax.workload import \
+    OgbgWorkload as JaxOgbgWorkload
+from algorithmic_efficiency.workloads.ogbg.ogbg_pytorch.workload import \
+    OgbgWorkload as PyTorchOgbgWorkload
+from algorithmic_efficiency.workloads.wmt.wmt_jax.workload import \
+    WmtWorkload as JaxWmtWorkload
+from algorithmic_efficiency.workloads.wmt.wmt_pytorch.workload import \
+    WmtWorkload as PyTorchWmtWorkload
+
+WORKLOADS = ['mnist', 'cifar', 'imagenet_resnet', 'imagenet_vit', 'wmt', 'ogbg']
+
+
+# Ideally we would match the types layer-wise, but for that we
+# have to ensure the exact same number of the types and that the
+# types of the weights of the same layer type actually match between
+# Jax and PyTorch, which is not always the case.
```

Comment:
is this because some layers fuse bias and weight together? can you put an example of the failure mode?